### ISSUE TITLE
5.1.12 macro expansion issue

### ISSUE THREAD

**Issue by hmlatapie** [Jermolene/TiddlyWiki5#2501]
<<macroname "parameter">> no longer work for expansions that result in URL's. Rather than open the expanded URL in a new tab like pre 5.1.12 versions, a new tiddler was opened with the URL as the title. Note, this bug manifested on node server version with nginx reverse proxy.

Moved back to 5.1.11 for now. 

Big fan of the TW... thanks!

**Comment by Jermolene** [comment]
Hi @hmlatapie thanks for the report. Can you give some example code that shows the problem? Do the URLs you are creating have single quotes or tildes (~) in them?

I think this may be related to an issue that has come to light whereby external URLs are checked incorrectly, but I'd like to understand the specifics of the issue here.

**Comment by hmlatapie** [comment]
Hi Jeremy,

Here’s one code sample that exhibited the problem: 

\define g(q:"")
[[google search ($q$)|https://www.google.com/?gws_rd=ssl#q=$q$]]
\end

And another:

\define y(q:"")
[[youtube search ($q$)|https://www.google.com/?gws_rd=ssl#q=site:youtube.com+$q$]]
\end

Thanks for looking into this. -Hugo

**Comment by sukima** [comment]
To save the eyes (code can be added in GitHub comments with triple back ticks):

```
\define g(q:"")
[[google search ($q$)|https://www.google.com/?gws_rd=ssl#q=$q$]]
\end
```

And another:

```
\define y(q:"")
[[youtube search ($q$)|https://www.google.com/?gws_rd=ssl#q=site:youtube.com+$q$]]
\end
```

**Comment by Jermolene** [comment]
Thanks @hmlatapie @sukima 

In 5.1.12 I tried the following example:

```
\define g(q:"")
[[google search ($q$)|https://www.google.com/?gws_rd=ssl#q=$q$]]
\end

<<g "Something>>
```

A link to `https://www.google.com/?gws_rd=ssl#q=Something` was correctly generated.

Are you doing something different?

**Comment by hmlatapie** [comment]
Interesting. When I click on the link produced by <<g something>> rather than open a tab with the google URL a new tiddler was opened. Perhaps there's an incompatibility with another tiddler on my system. 

I've been using Tiddlywiki for a few years now so perhaps it's time to start fresh and reload my non-system tiddlers. I don't use any special plug-ins.

**Comment by Jermolene** [comment]
Hi @hmlatapie can you give an example of the text you're passing as the parameter to the macro?

There is a known bug in 5.1.12 whereby URLs containing a tilde (~) or single quote won't autolink. The fix us to use `[ext[google search ($q$)|https://www.google.com/?gws_rd=ssl#q=$q$]]`.

**Comment by hmlatapie** [comment]
Here's the text I was passing: 

```
"Zoubin Ghahramani"
```

**Comment by Jermolene** [comment]
Ah, OK, it's the space character that is the problem.

Here's what I tried on http://tiddlywiki.com:

```
\define g(q:"")
[[google search ($q$)|https://www.google.com/?gws_rd=ssl#q=$q$]]
\end

Raw text generated by macro: <$text text=<<g "Zoubin Ghahramani">>/>

Wikified macro text <<g "Zoubin Ghahramani">>
```

The raw text generated by macro was:

```
[[google search (Zoubin Ghahramani)|https://www.google.com/?gws_rd=ssl#q=Zoubin Ghahramani]]
```

As you can see, it includes a space character, which is not technically a valid character for URLs. This causes the link handler not to recognise it as an external URL. In practice links containing spaces do work in HTML, because browsers are much looser than the specifications they implement.

All of this is a result of a fix that we applied for 5.1.2 to fix the bug #2255. The fix was in #2324.

It's clear now that the fix is problematic, and so I shall shortly simplify it to only ensure that the content of the pretty link looks like a URL at the start, and ignore any illegal URL characters.

**Comment by hmlatapie** [comment]
Thanks!
